pkg/apply/applier_test.go

```diff
@@ -303,28 +451,40 @@ func TestApplier(t *testing.T) {
 				},
 				{
 					eventType:       event.StatusType,
-					statusEventType: event.StatusEventResourceUpdate,
+					statusEventType: event.StatusEventCompleted,
 				},
 				{
-					eventType:       event.StatusType,
-					statusEventType: event.StatusEventCompleted,
+					// TODO(mortent): This doesn't seem correct. We shouldn't try
```

Comment:
In this case, we didn't try to prune this resource. We skip it. Maybe the `event.PruneSkipped` is confusing to you, we can change it to a more meaningful operation such as `event.UnPrune`.

Reply:
Yeah, so I realize we don't actually end up pruning it, but it just seems like in this case the resource shouldn't even be considered for pruning. My thought is that during apply, if a resource is part of the set of resources provided to the apply-task, we should include its `uid` in the `taskContext` so the resource during pruning is treated as if it actually was applied. WDYT?

Reply:
I agree that that the `uid` should be in included in the `taskContext`. This is some misbehavior introduced after adding the `canApply` in the ApplyTask. I think moving line https://github.com/kubernetes-sigs/cli-utils/blob/master/pkg/apply/task/apply_task.go#L150 ahead of the triggering of `CanApply` function can fix this.

Reply:
https://github.com/kubernetes-sigs/cli-utils/pull/303 is to fix the issue.